jsonschema2pojo-core/src/main/java/org/jsonschema2pojo/util/NameHelper.java

```diff
@@ -47,7 +47,36 @@ public String getBuilderTypeParameterName(JDefinedClass instanceClass) {
     }
 
     public String replaceIllegalCharacters(String name) {
-        return name.replaceAll(ILLEGAL_CHARACTER_REGEX, "_");
+        if (generationConfig.isAllowNonLatinNames()) {
+            return removeNotValidJavaIdentPartsStartingFrom(name, getValidJavaIdentifierStartIndex(name));
+        } else {
+            return name.replaceAll(ILLEGAL_CHARACTER_REGEX, "_");
+        }
+    }
+
+    private String removeNotValidJavaIdentPartsStartingFrom(String name, int validStartPosition) {
+        StringBuilder sb = new StringBuilder();
+        int length = name.length();
+        sb.append(name.charAt(validStartPosition));
```

Comment:
you all invalid symbols of the `name` while original algorithm replace all invalid characters with underscore. I'd like to preserve behaviour as it was, to make this one as an extended version of the previous

Reply:
This behavior is maintained as an option, underscores are then removed.
I don't see much profit in maintaining this behavior as such.

Reply:
Documentation for an option doesn't state, that behaviour is changed. Reading current documentation, I expect that more symbols would be accepted as valid, not only `[a-zA-Z0-9_$]`. Because of this I'd insist on maintaining the behaviour

Reply:
Done, please check